### PR TITLE
Apply #2137 to mixed precision

### PR DESCRIPTION
**Pull request by ye-luo**
Review after #3810

## Proposed changes
Apply the same choice in mixed precision. Use fixed size integer.
The number of k-shells is very sensitive to numerical errors thus skip unit tests.
The energy value should not be affected by k-shells and thus doesn't need a separate reference value.

## What type(s) of changes does this code introduce?
- Bugfix

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
Test this please

**Review by prckent — COMMENTED** [review]
Please restore the tests skipped in mixed precision with approximate checks. Very good to be able to have single references for ion-ion now though.

File: src/Particle/LongRange/tests/test_ewald3d.cpp [hunk 1]
```diff
@@ -29,25 +29,27 @@ TEST_CASE("ewald3d", "[lrhandler]")
...
+  CHECK(handler.Sigma == Approx(Lattice.LR_kc));
   handler.initBreakup(ref);
-  REQUIRE(handler.Sigma == Approx(std::sqrt(Lattice.LR_kc / (2.0 * Lattice.LR_rc))));
+  CHECK(handler.Sigma == Approx(std::sqrt(Lattice.LR_kc / (2.0 * Lattice.LR_rc))));
 
-  REQUIRE(handler.MaxKshell == 78);
-  REQUIRE(handler.LR_rc == Approx(2.5));
-  REQUIRE(handler.LR_kc == Approx(12));
+#if !defined(MIXED_PRECISION)
+  CHECK(handler.MaxKshell == 78);
```
[the remark this answers is not in the document]
Comment (ye-luo) [reply src/Particle/LongRange/tests/test_ewald3d.cpp:49]: Unfortunately approx only makes sense on float/double not on integers.

**Comment by prckent** [comment]
Yes, I meant a range by approx, +-2 or whatever is needed. Thanks.

**Review by prckent — APPROVED** [review]

**Pull request merged**